receiver/dockerstatsreceiver/metric_helper.go

```diff
@@ -71,3 +75,47 @@ func calculateMemoryPercent(limit uint64, usedNoCache uint64) float64 {
 	}
 	return 0.0
 }
+
+// calculateCPULimit calculate the number of cpus assigned to a container.
+//
+// Calculation is based on 3 alternatives by the following order:
+// - nanocpus:   if set by i.e docker run -cpus=2
+// - cpusetCpus: if set by i.e docker run -docker run -cpuset-cpus="0,2"
+// - cpuquota:   if set by i.e docker run -cpu-quota=50000
+//
+// See https://docs.docker.com/config/containers/resource_constraints/#configure-the-default-cfs-scheduler for background.
+func calculateCPULimit(hostConfig *ctypes.HostConfig) float64 {
+	var cpuLimit float64
+
+	switch {
+	case hostConfig.NanoCPUs > 0:
+		cpuLimit = float64(hostConfig.NanoCPUs) / 1e9
+	case hostConfig.CpusetCpus != "":
+		cpuLimit = parseCPUSet(hostConfig.CpusetCpus)
+	case hostConfig.CPUQuota > 0:
+		period := hostConfig.CPUPeriod
+		if period == 0 {
+			period = 100000 // Default CFS Period
+		}
+		cpuLimit = float64(hostConfig.CPUQuota) / float64(period)
+	}
+	return cpuLimit
```

Comment:
Isn't it better to not submit this metric if we can't find a limit? Rather than submitting a limit of 0 (zero) ?

Reply:
Wow, I was not aware of that possibility. Thanks a lot for taking the time to write the explanation for me. I think you are definitely right and we should keep this as it is. I'll resolve this comment.